### PR TITLE
Adds Deployment Data Source

### PR DESCRIPTION
**Pull request by itmecho**
I had a need to load a deployment from kubernetes to use the currently running container image in my terraform resource. This PR adds the `kubernetes_deployment` data source.

I ran the acceptance test against minikube and have used the data source with my current kubernetes setup (running on AWS) and it seems to be working as intended.

This is my first time working in this repo so any tips/errors I made would be great to know!

### CODE REVIEW

File: kubernetes/data_source_kubernetes_deployment.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func dataSourceKubernetesDeployment() *schema.Resource {
+	dsSchema := datasourceSchemaFromResourceSchema(resourceKubernetesDeployment().Schema)
+
+	addRequiredFieldsToSchema(dsSchema, "metadata")
+	addRequiredFieldsToSchema(dsSchema["metadata"].Elem.(*schema.Resource).Schema, "name")
```
Comment (itmecho) [inline kubernetes/data_source_kubernetes_deployment.go]: @sl1pm4t not sure is this is the best way to be doing this...

Comment (sl1pm4t) [reply kubernetes/data_source_kubernetes_deployment.go]: Ahh yeah, the heavily nested K8s resource schema's do complicate things a bit.
I'll have a look.

Comment (sl1pm4t) [reply kubernetes/data_source_kubernetes_deployment.go]: I think that's the best way to handle this for now. Good job.

**Comment by itmecho** [comment]
Awesome, Thanks!

**Pull request merged**